File: Part 2 - Regression/Section 6 - Polynomial Regression/Regression_Template/temp.py
#polynomial regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
Y = dataset.iloc[:, 2].values

# Splitting the dataset into the Training set and Test set
"""from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.2, random_state = 0)"""

# Feature Scaling
"""from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X_train = sc_X.fit_transform(X_train)
X_test = sc_X.transform(X_test)
sc_Y = StandardScaler()
Y_train = sc_Y.fit_transform(Y_train)"""


#Fitting regression model to dataset
#Create regressor here

#Predict with polynomial regression
regressor.predict(6.5)

#Visualizing regression results
X_grid=np.arange(min(X),max(X),0.1)
X_grid=X_grid.reshape((len(X),1))
plt.scatter(X,Y,color='red')
plt.plot(X_grid,regressor.predict(X_grid),color='blue')
plt.title('Truth or bluff (regression model)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 6 - Polynomial Regression/Regression_Template/temp.R
#Regression template

#read data
dataset = read.csv('Position_Salaries.csv')
dataset=dataset[,2:3]

#split test and training set
# install.packages('caTools')
# library(caTools)
# set.seed(123)
# split=sample.split(dataset$Purchased,SplitRatio=0.8)
# training_set=subset(dataset,split==TRUE)
# test_set=subset(dataset,split==FALSE)

#feature scaling
#training_set[,2:3]=scale(training_set[,2:3])
#test_set[,2:3]=scale(test_set[,2:3])

#Fitting regression model to dataset
#create regressor

#predict data using regression
y_pred=predict(regessor,data.frame(Level=6.5))


#visualize regression model
library(ggplot2)
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),
             colour='red')+
  geom_line(aes(x=dataset$Level,y=predict(regessor,newdata=dataset)),
            colour='blue')+
  ggtitle('True vs bluff (Regression model)')+
  xlab('Level')+
  ylab('Salary')

#visualize polynomial regression model more smooth
library(ggplot2)
X_grid=seq(min(dataset$Level),max(dataset$Level),0.1)
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),
             colour='red')+
  geom_line(aes(x=x_grid,y=predict(regessor,newdata=data.frame(Level=X_grid))),
            colour='blue')+
  ggtitle('True vs bluff (Regression model)')+
  xlab('Level')+
  ylab('Salary')























